src/main/java/com/box/sdk/BoxResource.java

```diff
@@ -22,30 +35,66 @@ public BoxResource(BoxAPIConnection api, String id) {
         this.id = id;
     }
 
+    /**
+     * @return Builds {@link Map} between String {@link #getType(Class)} and {@link BoxResource} type.
+     */
+    private static Map<String, Class<? extends BoxResource>> initResourceClassByType() {
+        Map<String, Class<? extends BoxResource>> result =
+                new ConcurrentHashMap<String, Class<? extends BoxResource>>();
+        result.put(getType(BoxFolder.class), BoxFolder.class);
+        result.put(getType(BoxFile.class), BoxFile.class);
+        result.put(getType(BoxComment.class), BoxComment.class);
+        result.put(getType(BoxCollaboration.class), BoxCollaboration.class);
+        result.put(getType(BoxTask.class), BoxTask.class);
+        result.put(getType(BoxTaskAssignment.class), BoxTaskAssignment.class);
+        result.put(getType(BoxUser.class), BoxUser.class);
+        result.put(getType(BoxGroup.class), BoxGroup.class);
+        result.put(getType(BoxGroupMembership.class), BoxGroupMembership.class);
+        result.put(getType(BoxEvent.class), BoxEvent.class);
+        result.put(getType(BoxWebHook.class), BoxWebHook.class);
+        return Collections.unmodifiableMap(result);
+    }
+
+    /**
+     * Resolves {@link BoxResourceType} for a provided {@link BoxResource} {@link Class}.
+     *
+     * @param clazz
+     *            {@link BoxResource} type
+     * @return resolved {@link BoxResourceType#value()}
+     */
+    public static String getType(Class<? extends BoxResource> clazz) {
```

Comment:
getType seems too generic.   I think getResourceType would make the code clearer.